src/app/core/messages/messages.service.ts

```diff
@@ -0,0 +1,146 @@
+import path from 'path';
+import { FilterQuery, PopulateOptions, Types } from 'mongoose';
+import { dbs, config, utilService } from '../../../dependencies';
+import { UserDocument } from '../user/types';
+import { PagingResults } from '../../common/mongoose/paginate.plugin';
+import { IMessage, MessageDocument, MessageModel } from './message.model';
+import {
+	DismissedMessageDocument,
+	DismissedMessageModel,
+	IDismissedMessage
+} from './dismissed-message.model';
+
+type PublishProvider = {
+	publish: (destination: string, message: unknown, retry: boolean) => void;
+};
+
+class MessagesService {
+	model: MessageModel;
+	dismissedModel: DismissedMessageModel;
+	publishProvider: PublishProvider;
+
+	constructor() {
+		this.model = dbs.admin.model('Message');
+		this.dismissedModel = dbs.admin.model('DismissedMessage');
+	}
+
+	create(user: UserDocument, doc: unknown): Promise<MessageDocument> {
+		const message = new this.model(doc);
+		message.creator = user._id;
+
+		return message.save();
+	}
+
+	read(
```

Comment:
Again, just not really sure about all these potential types, especially if they are just being coerced to string[] in the function

Reply:
The `populate` parameter is passed into mongoose's populate function.  The potential types are defined by mongoose.  We could limit it to the types that we actually use, but the intent was to create a consistent API across the services.

The coercion was needed due to the way that mongoose defines the typings for the populate method.  It is defined with two overloads, each of which accepts a different subset of the types.